main.tf

```diff
@@ -108,6 +108,14 @@ resource "google_vpc_access_connector" "main" {
   network        = google_compute_network.main.name
   region         = var.region
   max_throughput = 300
+  depends_on     = [time_sleep.wait_before_destroying_network]
+}
+
+# The google_vpc_access_connector resource creates some firewalls that sometimes takes a while to destroy
+# and causes errors when we try to destroy the VPC network (google_compute_network).
+resource "time_sleep" "wait_before_destroying_network" {
```

Comment:
future: If the wait time on this is inconsistent and could range longer, consider using a polling technique such as https://github.com/GoogleCloudPlatform/terraform-dynamic-javascript-webapp/blob/db3eb93c495d4db9456e9154314479cad9a699e0/infra/main.tf#L302